### PR TITLE
lints + flow

### PR DESCRIPTION
**Pull request by alexeuler**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/reactrails-react-native-client/24)

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -43,3 +45,7 @@ rules:
   flowtype/type-id-match:
   - 1
   - "^([A-Z][a-z0-9]+)+Type$"
+
+  react/jsx-filename-extension:
+    - 1
+    - extensions: [".js", ".jsx"]
```
Comment (justin808) [inline .eslintrc.yml:51]: @robwise @alexfedoseev How about we put this into the shakacode linter?

File: .eslintrc.yml [hunk 2]
```diff
@@ -21,6 +21,8 @@ settings:
     onlyFilesWithFlowAnnotation: true
 globals:
   __DEV__: true
+  ReactClass: true
+  ReactElement: true
```
Comment (justin808) [inline .eslintrc.yml:25]: @robwise @alexfedoseev Should these go into the shakacode linter?

File: .eslintrc.yml (hunk 1 above)
Comment (robwise) [reply .eslintrc.yml:51]: @justin808 I commented on the styleguide repo, what's the reasoning?

**Review by justin808 — COMMENTED** [review]
@robwise are you suggesting that it's better to mark files with jsx with a `.jsx` extension?

**Comment by robwise** [comment]
> @robwise are you suggesting that it's better to mark files with jsx with a .jsx extension?

@justin808 Yes that's what we've always been doing. If it has JSX in it, it's a jsx file. If it doesn't, it's a JS file.

File: .eslintrc.yml (hunk 2 above)
Comment (alex35mil) [reply .eslintrc.yml:25]: Isn't it React Native specific?

File: .eslintrc.yml (hunk 1 above)
Comment (alex35mil) [reply .eslintrc.yml:51]: @justin808 It should be:

```yml
react/jsx-filename-extension:
  - 1
  - extensions: [".js"]

# or

react/jsx-filename-extension:
  - 1
  - extensions: [".jsx"]

```

In general I'm fine w/ either, but as we already have a big codebase, that's using `.jsx`, I'd stay were we are as there's no reason to spend time on this change.

File: .eslintrc.yml (hunk 2 above)
Comment (robwise) [reply .eslintrc.yml:25]: These aren't React-Native specific, these are used when typing React with flow. If you use [eslint-plugin-flowtype](https://github.com/gajus/eslint-plugin-flowtype), defining these as globals isn't necessary and these can be removed. I would also recommend using the [recommend configuation](https://github.com/gajus/eslint-plugin-flowtype#recommended).

**Comment by justin808** [comment]
@robwise @alexfedoseev as far as I can tell, the FB examples for react and react-native don't use the jsx extension.

**Comment by robwise** [comment]
@justin808 We shouldn't deviate from the Airbnb styleguide unless we have a good reason. We would have to change an unbelievably monstrous number of files on F&G if we implement this. It's a good rule IMO anyway.

File: .eslintrc.yml (hunk 2 above)
Comment (alex35mil) [reply .eslintrc.yml:25]: This is crazy train, I [can't find](https://github.com/facebook/flow/blame/master/lib/react.js) anything about `ReactElement` here. AFAIR I had issues w/ it and bunch of examples were using `React$Element`—it is in the defs, but it wasn't work either. Then I found [this post](http://technologyadvice.github.io/eradicate-runtime-errors-in-react-with-flow/), where stated that `ReactElement` is removed from public API and `React.Element` should be used instead. The funny thing is that it worked and still works, but: there is no `Element` property on `React` object lol.

**Pull request merged**